src/sqlfluff/dialects/dialect_sparksql.py

```diff
@@ -981,7 +972,7 @@ class InsertStatementSegment(BaseSegment):
     https://spark.apache.org/docs/latest/sql-ref-syntax-dml-insert-overwrite-table.html
     """
 
-    type = "insert_table_statement"
+    type = "insert_statement"
```

Comment:
Fixed a `type` mismatch with the ANSI segment type. Did not use inheritance because ANSI segment uses `parse_grammar` but this one doesn't.